pkg/server/v2/server.go

```diff
@@ -161,15 +161,12 @@ func createResponse(resp *cache.Response, typeURL string) (*discovery.DiscoveryR
 				Value:   resp.MarshaledResources[i],
 			}
 		} else {
-			marshaledResource, err := cache.MarshalResource(resp.Resources[i])
+			marshaledResource, err := cache.MarshalResource(resp.Resources[i], typeURL)
```

Comment:
should `discovery.DiscoveryResponse.TypeUrl` below be updated as well?

Reply:
i see...trying that out.

Reply:
There;s still some issue. From envoy logs 
```
[2020-04-17 19:18:00.722][1209][debug][config] [source/common/config/grpc_mux_impl.cc:136] Received gRPC message for type.googleapis.com/envoy.config.cluster.v3.Cluster at version v0
[2020-04-17 19:18:00.722][1209][warning][config] [source/common/config/grpc_mux_impl.cc:139] Ignoring the message for type URL type.googleapis.com/envoy.config.cluster.v3.Cluster as it has no current subscribers.
```